src/emulator/functionsEmulatorShared.ts

```diff
@@ -4,13 +4,12 @@ import * as os from "os";
 import * as path from "path";
 import * as express from "express";
 import * as fs from "fs";
+
 import { Constants } from "./constants";
 import { InvokeRuntimeOpts } from "./functionsEmulator";
+import { FunctionsPlatform } from "../deploy/functions/backend";
 
-export enum EmulatedTriggerType {
-  BACKGROUND = "BACKGROUND",
-  HTTPS = "HTTPS",
-}
+export type SignatureType = "http" | "event" | "cloudevent";
```

Comment:
nit: Why string instead of enum?

Reply:
I learned to prefer string unions from @joehan and @inlined. Found some interesting discussions on enums vs union type [online](https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant) - do you have preference for enums?

Reply:
I'm fine either way, but you know there's gonna be a nit question when you change things :)